Applications/Spire/Include/Spire/Ui/CustomQtVariants.hpp

```diff
@@ -124,21 +125,105 @@ namespace Spire {
   /** Returns the text representation of a TimeInForce. */
   const QString& displayText(Nexus::TimeInForce time_in_force);
 
+  /**
+   * Returns a TimeInForce that's represented by the given string.
+   * @param string The string representation.
+   * @returns An initialized optional iff the string represents a TimeInForce.
+   */
+  boost::optional<Nexus::TimeInForce> to_time_in_force(const QString& string);
+
   /** Returns the text representation of a Side. */
   const QString& displayText(Nexus::Side side);
 
+  /**
+   * Returns a Side that's represented by the given string.
+   * @param string The string representation.
+   * @returns An initialized optional iff the string represents a Side.
+   */
+  boost::optional<Nexus::Side> to_side(const QString& string);
```

Comment:
Use static polymorphism for this, instead of 100 ```to_this```/```to_that```, have one single:

    template<typename T>
    boost::optional<T> from_string(const QString& source) = delete;

And then provide specializations of it. This allows ```from_string``` to be used polymorphically in a generic context.